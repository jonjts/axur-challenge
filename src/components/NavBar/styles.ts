import styled from 'styled-components';

export const Container = styled.header`
    width: 100%;
    height: 80px;
    display: flex;
    align-items: center;
    background-color: ${({ theme: { colors: { background } } }) => background};
    box-shadow: 0px 11px 15px -3px rgba(0,0,0,0.21);
    position: relative;

    @media (min-width: 0px) {
        justify-content: center;
     }

    // Large devices (desktops, 992px and up)
    @media (min-width: 992px) { 
        justify-content: flex-start;
        padding: ${({ theme: { spacings: { lg } } }) => `0 ${lg.percent} 0 ${lg.percent}`};
     }
`;

export const Brand = styled.img`
    height: 40px;
    width: 40px;
    padding: 8px;
    background-color: ${({ theme: { colors: { primary } } }) => primary};
    border-radius: ${({ theme: { border } }) => border};
`;
