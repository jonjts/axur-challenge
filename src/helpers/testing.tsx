import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import ThemeProvider from '../providers/Theme';
import InspectionProvider from '../providers/Inspection'

export const putProviders = (component: JSX.Element) => (
    <ThemeProvider>
        <InspectionProvider>
            {component}
        </InspectionProvider>
    </ThemeProvider>
);

export const matchSnapshotWithProviders = (component: JSX.Element) => {
    const tree = renderer.create(putProviders(component)).toJSON();
    expect(tree).toMatchSnapshot();
    return tree;
};

export const renderWithProviders = (component: JSX.Element) => {
    return render(putProviders(component));
};
