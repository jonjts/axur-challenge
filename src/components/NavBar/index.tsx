import React from 'react';

import { Container, Brand } from './styles';

import BrandImage from '../../assets/images/brand.svg';

const NavBar: React.FC = () => (
    <Container>
        <a href='/' title='Ir para home'>
            <Brand src={BrandImage} alt='Logo do app: desafio Axur' />
        </a>
    </Container >
);

export default NavBar;