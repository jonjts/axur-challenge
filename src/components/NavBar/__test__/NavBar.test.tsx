import { matchSnapshotWithProviders, renderWithProviders } from '../../../helpers/testing';

import NavBar from '../index';

describe('NavBar', () => {
  test('match with snapshot', () => {
    matchSnapshotWithProviders(<NavBar />);
  });

  test('renders correctly', () => {
    renderWithProviders(<NavBar />);
  });
});
