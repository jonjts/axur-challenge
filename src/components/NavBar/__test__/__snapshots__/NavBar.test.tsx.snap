// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar match with snapshot 1`] = `
<header
  className="sc-gsTCUz iUNJdd"
>
  <a
    href="/"
    title="Ir para home"
  >
    <img
      alt="Logo do app: desafio Axur"
      className="sc-dlfnbm ebWWjJ"
      src="brand.svg"
    />
  </a>
</header>
`;
